scylla-proxy/src/proxy.rs

```diff
@@ -0,0 +1,766 @@
+                                self.connection_close_notifier.try_recv().is_err() {
+                                warn!("Response processor had exited");
+                            }
+                            break
+                        }
+                    }
+                }
+                _terminate = self.terminate_notifier.recv() => break,
+
+                _close = self.connection_close_notifier.recv() => break,
+            }
+        }
+        self.exit("sender_to_driver");
+    }
+
+    async fn sender_to_cluster(
+        mut self,
+        mut write_half: (impl AsyncWrite + Unpin),
+        mut requests_rx: mpsc::UnboundedReceiver<RequestFrame>,
+        error_propagator: ErrorPropagator,
+    ) {
+        loop {
+            tokio::select! {
+                request = requests_rx.recv() => {
+                    match request {
+                        Some(request) => {
+                            debug!("Sending Proxy ({}) -> Cluster ({}) frame. opcode: {:?}.", self.proxy_addr, self.real_addr, &request.opcode);
+                            if request.write(&mut write_half).await.is_err() {
+                                if self.terminate_notifier.try_recv().is_err() &&
+                                    self.connection_close_notifier.try_recv().is_err() {
+                                    warn!("Node {} dropped connection", self.real_addr);
+                                    let _ = error_propagator.send(WorkerError::NodeDisconnected(self.real_addr).into());
+                                }
+                                break;
+                            }
+                        }
+                        None => {
+                            if self.terminate_notifier.try_recv().is_err() &&
+                                self.connection_close_notifier.try_recv().is_err() {
+                                warn!("Request processor had exited");
+                            }
+                            break
+                        }
+                    }
+                }
+                _terminate = self.terminate_notifier.recv() => break,
+
+                _close = self.connection_close_notifier.recv() => break,
+            }
+        }
+        self.exit("sender_to_cluster");
+    }
+
+    async fn request_processor(
+        mut self,
+        mut requests_rx: mpsc::UnboundedReceiver<RequestFrame>,
+        driver_tx: mpsc::UnboundedSender<ResponseFrame>,
+        cluster_tx: mpsc::UnboundedSender<RequestFrame>,
+        connection_no: usize,
+        request_rules: Arc<RwLock<Vec<RequestRule>>>,
+        connection_close_signaler: ConnectionCloseSignaler,
+    ) {
+        'mainloop: loop {
+            tokio::select! {
+                request = requests_rx.recv() => {
+                    match request {
+                        Some(request) => {
+                            let ctx = EvaluationContext{
```

Comment:
Looks like another bug in rustfmt - it forgot to add a space between `EvaluationContext` and `{`.

Reply:
I suspect it is due to being inside a macro (`tokio::select!`)